src/main/java/org/alfresco/repo/policy/EventBehaviour.java

```diff
@@ -0,0 +1,242 @@
+        ParameterCheck.mandatory("Method", method);
+
+        this.eventProducer = eventProducer;
+        this.endpointUri = endpointUri;
+        this.instance = instance;
+        this.method = method;
+    }
+
+
+    @Override
+    public String toString()
+    {
+        return "Java method[class=" + instance.getClass().getName() + ", method=" + method + "]";
+    }
+    
+    @SuppressWarnings("unchecked")
+	public synchronized <T> T getInterface(Class<T> policy) 
+	{
+	    ParameterCheck.mandatory("Policy class", policy);
+	    Object proxy = proxies.get(policy);
+	    if (proxy == null)
+	    {
+	        InvocationHandler handler = getInvocationHandler(eventProducer, endpointUri, instance, method, policy);
+	        proxy = Proxy.newProxyInstance(policy.getClassLoader(), new Class[]{policy}, handler);
+	        proxies.put(policy, proxy);
+	    }
+	    return (T)proxy;
+	}
+
+    /**
+     * Gets the Invocation Handler.
+     * 
+     * @param <T>  the policy interface class
+     * @param instance  the object instance
+     * @param method  the method name
+     * @param policyIF  the policy interface class  
+     * @return  the invocation handler
+     */
+    <T> InvocationHandler getInvocationHandler(EventProducer eventProducer, String endpointUri, Object instance, String method, Class<T> policyIF)
+    {
+        Method[] policyIFMethods = policyIF.getMethods();
+        if (policyIFMethods.length != 1)
+        {
+            throw new PolicyException("Policy interface " + policyIF.getCanonicalName() + " must have only one method");
+        }
+
+        try
+        {
+            Class instanceClass = instance.getClass();
+            Method delegateMethod = instanceClass.getMethod(method, (Class[])policyIFMethods[0].getParameterTypes());
+            return new JavaMethodInvocationHandler(eventProducer, endpointUri, this, delegateMethod);
+        }
+        catch (NoSuchMethodException e)
+        {
+            throw new PolicyException("Method " + method + " not found or accessible on " + instance.getClass(), e);
+        }
+    }    
+    
+    /**
+     * Java Method Invocation Handler
+     *
+     */
+    private static class JavaMethodInvocationHandler implements InvocationHandler
+    {
+        private EventProducer eventProducer;
+        private String endpointUri;
+        private EventBehaviour behaviour;
+        private Method delegateMethod;
+        
+        /**
+         * Constuct.
+         * 
+         * @param behaviour  the java behaviour
+         * @param delegateMethod  the method to invoke
+         */
+        private JavaMethodInvocationHandler(EventProducer eventProducer, String endpointUri, EventBehaviour behaviour, Method delegateMethod)
+        {
+            this.eventProducer = eventProducer;
+            this.endpointUri = endpointUri;
+            this.behaviour = behaviour;
+            this.delegateMethod = delegateMethod;
+        }
+
+        /* (non-Javadoc)
```

Comment:
(non-Javadoc) is just a waste of screen space and does not provide any information that is not available in an IDE.  Please turn off this auto-generation in IDE.